@ecomplus/storefront-components/src/js/ProductCard.js

```diff
@@ -146,6 +156,11 @@ export default {
       }
     },
 
+    toggleFavorite () {
+      toggleFavorite(this.body._id, this.ecomPassport)
+      this.isFavorite = checkFavorite(this.body._id, this.ecomPassport)
```

Comment:
```suggestion
      this.isFavorite = toggleFavorite(this.body._id, this.ecomPassport)
```

o próprio `toggleFavorite` já começa checando o se o produto está entre os favoritos ou não, basta ele retornar isso, você repetir o `checkFavorite` depois é redundante..